napari/utils/events/dataclass.py

```diff
@@ -389,6 +419,23 @@ def convert_fields_to_properties(cls: Type[C]) -> Type[C]:
     return cls
 
 
+def update_from_dict(self, values, compare_fun=compare):
+    if isinstance(values, self.__class__):
+        values = values.asdict()
+    if not isinstance(values, dict):
+        raise ValueError(f"Unsupported update from {type(values)}")
+    if all(compare(values[k], v) for k, v in self.asdict().items()) == values:
+        return
+
+    self.events.block()
+
+    for key, value in values.items():
+        setattr(self, key, value)
```

Comment:
I'm fine with this for now, but we might want to revisit if there are performance issues